board/stm32h7/clock.h

```diff
@@ -17,13 +17,35 @@ APB4 per: 60MHz
 PCLK1: 60MHz (for USART2,3,4,5,7,8)
 */
 
-void clock_init(void) {
+bool failed_clock_init = false;
+
+void clock_init(bool allow_vos3) {
   //Set power mode to direct SMPS power supply(depends on the board layout)
   register_set(&(PWR->CR3), PWR_CR3_SMPSEN, 0xFU); // powered only by SMPS
-  //Set VOS level (VOS3 to 170Mhz, VOS2 to 300Mhz, VOS1 to 400Mhz, VOS0 to 550Mhz)
+
+  // Set VOS level (VOS3 to 170Mhz, VOS2 to 300Mhz, VOS1 to 400Mhz, VOS0 to 550Mhz)
   register_set(&(PWR->D3CR), PWR_D3CR_VOS_1 | PWR_D3CR_VOS_0, 0xC000U); //VOS1, needed for 80Mhz CAN FD
-  while ((PWR->CSR1 & PWR_CSR1_ACTVOSRDY) == 0);
-  while ((PWR->CSR1 & PWR_CSR1_ACTVOS) != (PWR->D3CR & PWR_D3CR_VOS)); // check that VOS level was actually set
+
+  // Wait until VOS level is reached. If flashed through DFU right after power-on, the CR3 register is locked, so this won't work.
```

Comment:
can you add something about how this only affects some chips with an older bootloader? and link to a reference from ST if it exists